### ISSUE TITLE
[Documentation] Fix section 4.8 based on Professor's Feedback

### ISSUE THREAD

**Issue by JConcepcion01** [uprm-inso4116-2024-2025-s1/semester-project-trolley-tracker-app#193]
Address notes made by the professor on the Documentation. Please refer to the following document for more details: https://docs.google.com/document/d/1iqslezp9a2G_6m-8B7yVRagj3M7xjivETaEn-bo2t2M/edit?usp=sharing

**Comment by yamayrie** [comment]
Developer’s internal testing is needed to ensuring that the Trolley Tracker app satisfies user needs. This section shows how developers iteratively tested and improved the app by assuming the roles of possible users.

**Predicting User Behavior**
Developers used the app while assuming the roles of administrators, professors, and students so we can better understand user challenges. Each role offered a different view:
- **Students:** Focused on tracking in real time and getting alerts about delays in the route.
- **Faculty:** Studied the dependability of sticking to schedules and route planning.
- **Administrators:** To improve system efficiency, we looked at route allocations and metrics reporting.
The team was able to pinpoint issues including poor user interfaces, delayed messages, and imprecise tracking thanks to these simulations.

**Making Adjustments**
Theme 1: Interface Design
- During student role simulations, developers discovered that the interface was difficult to use, so they redesigned it with bigger route icons and clearer labels.

Theme 2: Notifications
- Personalized alert settings were implemented because developers realized that passengers needed customizable notifications.

Theme 3: Real-Time Tracking
- Developers found irregularities in location updates by testing as students, which led to improvements in the GPS data synchronization.
By physically utilizing the trolley system and examining the surroundings, the development team used "Digital Anthropology" to get personal experience in the topic. For example, they pointed out how crucial it is to notify trolleys to stop, which resulted in improved notice prompts and more understandable user instructions. This practical method helped close the gap between theoretical concepts and practical implementation.

Developers found and fixed major issues by testing as end users, which enhanced the app's reliability and usefulness. To make sure constant alignment with user needs, we want to set up an ongoing internal feedback loop in which developers test modifications in simulated scenarios.
